lib/Doctrine/DBAL/Types/WellKnownTextConverter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\DBALException;
+
+/**
+ * Does working with this column require SQL conversion functions?
+ *
+ * This is a metadata function that is required for example in the ORM.
+ * Usage of {@link convertToDatabaseValueSQL} and
+ * {@link convertToPHPValueSQL} works for any type and mostly
+ * does nothing. This method can additionally be used for optimization purposes.
+ *
+ * @author Roman Borschel <roman@code-factory.org>
+ * @author Benjamin Eberlei <kontakt@beberlei.de>
+ * @since  2.0
+ */
+interface WellKnownTextConverter
+{
+    /**
+     * Modifies the SQL expression (identifier, parameter) to convert to a database value.
+     *
+     * @param string                                    $sqlExpr
+     * @param \Doctrine\DBAL\Platforms\AbstractPlatform $platform
+     *
+     * @return string
+     */
+    public function convertFromWellKnownTextSQL(string $sqlExpr, AbstractPlatform $platform): string;
```

Comment:
I deliberately picked other method names, so that I could add type hinting.

Reply:
Actually, since this is targeting 2.7 which will probably require PHP 7.2 (according to current policy of _releasing for latest stable PHP_), you could add parameter typehints without breaking public API. Not return type though, those could only be done later in 3.0.

Reply:
I see the RC2 is released, so maybe this will be quite soon

Reply:
ETA @ end of november.

Reply:
Spaces between return types. The return types will stay in the interface, right? :-)

Reply:
> Not return type though, those could only be done later in 3.0.

@SenseException not completely sure what @Majkl578 meant with that.

Reply:
Those methods are new, so my guess is, that type widening isn't relevant here and the interface's API with the return type can stay as it is. Return types aren't covered in type widening and he just mentioned that.

Reply:
> The return types will stay in the interface

Not in 2.x, adding return types to ancestors is a BC break, only parameter types are acceptable (and even those only as of PHP 7.2).

As I outlined on Slack, it may be easiest to just not add types at all now, just introduce the interface and add types later when master requires PHP 7.2+.